From 5743f52a3578cf3c639eff968f5e695e58b15fd3 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?SimaMoto=2CRy=C5=8DTa?= <liangtai.s16@gmail.com>
Date: Thu, 30 Jan 2014 21:52:55 +0900
Subject: [PATCH] Fix option handling

Fix configuration failure on !WANT_PLAYER.

Do not install twice dynamic libray and header on UNIX && !APPLE.

Do not generate static library on !(WANT_STATIC || WANT_PLAYER).
---
 CMakeLists.txt     | 18 ++++++++++--------
 src/CMakeLists.txt | 18 ++++++++++++++----
 2 files changed, 24 insertions(+), 12 deletions(-)

diff --git CMakeLists.txt CMakeLists.txt
index 39995d4..e4b5c97 100644
--- CMakeLists.txt
+++ CMakeLists.txt
@@ -54,13 +54,15 @@ SET(WILDMIDI_CFG "wildmidi.cfg")
 IF(NOT WIN32 AND NOT APPLE)
 
     # go looking for available sound packages for WildMIDI player
-    IF(WANT_PLAYER AND WANT_ALSA)
-        FIND_PACKAGE(ALSA REQUIRED)
-    ELSEIF(WANT_PLAYER AND WANT_OSS)
-        FIND_PACKAGE(OSS REQUIRED)
-    ELSE(WANT_PLAYER AND WANT_ALSA)
-        MESSAGE(FATAL_ERROR "You must choose either ALSA or OSS.") 
-    ENDIF(WANT_PLAYER AND WANT_ALSA)
+    IF(WANT_PLAYER)
+        IF(WANT_ALSA)
+            FIND_PACKAGE(ALSA REQUIRED)
+        ELSEIF(WANT_OSS)
+            FIND_PACKAGE(OSS REQUIRED)
+        ELSE(WANT_ALSA)
+            MESSAGE(FATAL_ERROR "You must choose either ALSA or OSS.")
+        ENDIF(WANT_ALSA)
+    ENDIF(WANT_PLAYER)
     
     # find our math lib
     FIND_LIBRARY(M_LIBRARY m REQUIRED)
diff --git src/CMakeLists.txt src/CMakeLists.txt
index 75fa6fb..40fa17c 100644
--- src/CMakeLists.txt
+++ src/CMakeLists.txt
@@ -17,16 +17,18 @@ SET(LIBRARY_OUTPUT_PATH "${CMAKE_BINARY_DIR}")
 
 MESSAGE(STATUS "${CMAKE_BINARY_DIR}")
 
+IF(WANT_STATIC)
 # actual target:
 ADD_LIBRARY(wildmidi_static STATIC 
     ${wildmidi_library_SRCS}
     )
 
-INSTALL(TARGETS wildmidi_static DESTINATION lib)
+#INSTALL(TARGETS wildmidi_static DESTINATION lib)
 
 TARGET_LINK_LIBRARIES(wildmidi_static 
     ${M_LIBRARY}
     )
+ENDIF(WANT_STATIC)
 
 IF(WIN32)
     SET(LIBRARY_DYN_NAME "wildmidi_dynamic")
@@ -36,16 +38,18 @@ ELSE(WIN32)
     SET(LIBRARY_STATIC_NAME "WildMidi")
 ENDIF(WIN32)
 
+IF(WANT_STATIC)
 SET_TARGET_PROPERTIES(wildmidi_static PROPERTIES 
     OUTPUT_NAME ${LIBRARY_STATIC_NAME} CLEAN_DIRECT_OUTPUT 1
     )
+ENDIF(WANT_STATIC)
 
 ADD_LIBRARY(wildmidi_dynamic SHARED
     ${wildmidi_library_SRCS}
     )
 
-INSTALL(TARGETS wildmidi_dynamic DESTINATION lib)
-INSTALL(FILES ../include/wildmidi_lib.h DESTINATION include)
+#INSTALL(TARGETS wildmidi_dynamic DESTINATION lib)
+#INSTALL(FILES ../include/wildmidi_lib.h DESTINATION include)
 
 TARGET_LINK_LIBRARIES(wildmidi_dynamic 
     ${M_LIBRARY}
@@ -77,6 +81,7 @@ ELSEIF(CMAKE_GENERATOR STREQUAL "Xcode")
     ENDIF(WANT_STATIC)
 ENDIF(MSVC)
 
+IF(WANT_PLAYER)
 # Setup our wildmidi player
 SET(wildmidi_executable_SRCS
     ${GETOPT}
@@ -97,11 +102,14 @@ TARGET_LINK_LIBRARIES(wildmidi
     ${COREAUDIO_LIBRARY}
     ${COREFOUNDATION_LIBRARY}
     )
+ENDIF(WANT_PLAYER)
 
 # add install target:
 IF (UNIX AND NOT APPLE)
+IF(WANT_STATIC)
     # install our libraries
     INSTALL(TARGETS wildmidi_static DESTINATION "lib/${CMAKE_LIBRARY_ARCHITECTURE}")
+ENDIF(WANT_STATIC)
     INSTALL(TARGETS wildmidi_dynamic DESTINATION "lib/${CMAKE_LIBRARY_ARCHITECTURE}")
     
     # install our player if asked for
-- 
1.8.5.2

